plugins/bundle/plugin_test.go

```diff
@@ -6522,6 +6522,68 @@ func TestPluginManualTriggerWithTimeout(t *testing.T) {
 	}
 }
 
+func TestPluginManualTriggerWithServerError(t *testing.T) {
```

Comment:
This test triggers the race condition.
My investigation led to the conclusion that this is triggered by having the bundle with the "periodic" trigger mode in the test. 
The downloader.oneShot function will be triggered by the explicitly trigger method and also by the downloader.loop function.
The access to the shared variables (f.ex. etag, client) in Downloader is not synchronized and leads to the race condition.
A possible solution is to only trigger bundle dowloads with "manual" trigger mode in the trigger method.
F.ex. like this (in bundle.Plugin.Trigger)
```
	for name, dl := range p.downloaders {
		downloaders[name] = dl
		p.cfgMtx.RLock()
		triggerMode := p.config.Bundles[name].Trigger
		p.cfgMtx.RUnlock()

		if triggerMode != nil && *triggerMode == plugins.TriggerManual {
			downloaders[name] = dl
		}
	}
```
This would also solve the question if Trigger should return errors for bundles with "periodic" trigger mode.

Reply:
@ashutosh-narkar 
Do you have a recommendation how to proceed with this?
To make the test green, I could remove the configuration for the bundle in "periodic" trigger mode but I think this might be a problem in the bundle plugin that should be addressed.

Reply:
I think having only a manual trigger for the test should be fine.